packages/redux-dynamic-modules-react/src/DynamicModuleLoader.tsx

```diff
@@ -118,60 +75,71 @@ class DynamicModuleLoaderImpl extends React.Component<
         }
     }
 
-    private _addModules(): void {
-        const { createStore, modules } = this.props;
-        if (!this._store) {
-            if (createStore) {
-                this._store = createStore();
-                this._providerInitializationNeeded = true;
-            } else {
-                throw new Error(
-                    "Store could not be resolved from React context"
+    public render(): React.ReactNode {
+        if (this.state.readyToRender) {
+            if (this._providerInitializationNeeded) {
+                return (
+                    <Provider store={this._store}>
+                        {/* We just rendered the provider, so now we need to render
+                        DML again. This one will add the modules */}
+                        <DynamicModuleLoader {...this.props} />
+                    </Provider>
                 );
             }
-        } else {
-            // We will add modules dynamically and due to github issue https://github.com/Microsoft/redux-dynamic-modules/issues/27#issuecomment-464905893
-            // The very first render will not get latest state, to fix that we will need to get latest state from store directly on first render
-            this._getLatestState = ReactReduxContext;
+
+            return this._renderLoader();
         }
 
-        this._addedModules = this._store.addModules(modules);
+        return null;
     }
 
-    private _renderWithReactReduxContext = () => {
-        const { store } = this.props;
-        // store.getState is important here as we don't want to use storeState from the provided context
+    /**
+     * Render a Redux provider
+     */
+    private _renderLoader(): React.ReactNode {
+        if (this.props.reactReduxContext == null) {
+            const message =
+                "Tried to render DynamicModuleLoader, but no ReactReduxContext was provided";
+            console.error(message);
+
+            throw new Error(message);
+        }
+
+        // Memoize the context if it has changed upstream
+        if (this.props.reactReduxContext !== this._memoizedRRContext) {
+            this._memoizedRRContext = {
+                ...this.props.reactReduxContext,
+                storeState: this.props.reactReduxContext.store.getState(),

```

Comment:
Why are we memoizing state only when context changes?

Reply:
I overwrite state now, every time on render. Not sure if that is the right approach either, but we dont want to be re-creating this object everytime

Reply:
If the state changes above, then we will get a new context object here.

See: https://github.com/reduxjs/react-redux/blob/162b81a3fffe75ff6181b711777067ba6e63a34b/src/components/Provider.js#L59

this.setState({...}) returns a new object